core/primitives/src/test_utils.rs

```diff
@@ -169,24 +150,28 @@ impl SignedTransaction {
         .sign(signer)
     }
 
-    /// Explicitly create v1 transaction to test in cases where errors are expected.
+    /// Creates a v1 transaction. Use None `nonce_index` for access key and Some for gas key.
```

Comment:
```suggestion
    /// Creates a v1 transaction.
    ///
    /// Use `None` `nonce_index` for a transaction signed by an access key
    /// and `Some` for a transaction signed by a gas key.
```

That said, you already have a `TransactionKey` type, why not pass that directly? It is more type safe and way more self-explanatory than a generic `Some/None` argument.